test/test-parse-typescript-class.cpp

```diff
@@ -1933,6 +1933,80 @@ TEST_F(test_parse_typescript_class,
         }));
   }
 }
+
+TEST_F(test_parse_typescript_class,
+       class_keyword_with_escape_sequence) {
```

Comment:
Because this is strange behavior, I think it deserves a comment. Something like this:
```c++
// NOTE[typescript-constructor-escape]: The TypeScript compiler complains if
// 'constructor' in a class contains an identifier escape sequence, despite
// escapes being allowed in vanilla JavaScript. TypeScript does not complain if
// there is more than one sequence, though.
```

And for the other tests (no_diag_for_more_than_one_escape and class_keyword_with_escape_sequence_legal_in_js), you can reference:
```c++
// See NOTE[typescript-constructor-escape].
```

Reply:
And maybe reference the note in the implementation code too, 'cus why not? =]

Reply:
Resolved,
1. Added NOTE above test.
2. Reference 'See NOTE' right above implementation if condition.